### PR TITLE
Add workspace analyzer and user docs

### PR DESCRIPTION
**Pull request by chase6305**
## Description

This PR introduces a comprehensive **Workspace Analyzer** module for robotic workspace analysis in EmbodiChain. The module provides both joint space and Cartesian space analysis capabilities to help understand robot reachability, workspace volume, and performance metrics.

### Key Features Added:

- **Modular Configuration System**: Type-safe configuration classes for sampling, caching, visualization, constraints, and metrics
- **Multiple Sampling Strategies**: UniformSampler, RandomSampler, HaltonSampler, SobolSampler, LatinHypercubeSampler, ImportanceSampler, and GaussianSampler
- **Comprehensive Documentation**: Complete user guides and API documentation for all components
- **Factory Pattern Implementation**: Unified sampler creation and management system
- **Performance Optimization**: Caching mechanisms and efficient sampling algorithms

### Components Implemented:

1. **Core Workspace Analyzer** (`workspace_analyzer.py`)
   - Joint space and Cartesian space analysis modes
   - Configurable analysis parameters
   - Integration with simulation manager and robot models

2. **Configuration System** (`configs/`)
   - `WorkspaceAnalyzerConfig`: Main configuration dataclass
   - `SamplingConfig`: Sampling strategy configuration
   - `CacheConfig`: Result caching configuration
   - `VisualizationConfig`: Visualization settings
   - `MetricConfig`: Metric computation settings
   - `DimensionConstraint`: Workspace constraints

3. **Sampling Strategies** (`samplers/`)
   - Multiple statistical and quasi-random sampling methods
   - Factory pattern for sampler creation
   - Performance-optimized implementations

4. **Supporting Modules**
   - Caching system for performance optimization
   - Constraint handling for joint limits and collision detection
   - Visualization utilities for analysis results
   - Metric computation for workspace characteristics

### Documentation Added:

- **Main Documentation** (`workspace_analyzer.md`): Comprehensive usage guide with examples
- **Configuration Guide** (`configs.md`): Detailed configuration system documentation
- **Sampling Strategies** (`samplers.md`): In-depth sampling method descriptions
- **Visualization Options** (`visualizers.md`): Visualization configuration guide
- **Caching System** (`caches.md`): Performance optimization documentation
- **Constraint Handling** (`constraints.md`): Constraint system documentation
- **Metric Computation** (`metrics.md`): Analysis metrics documentation
[dexsim.tar.gz](https://github.com/user-attachments/files/23879535/dexsim.tar.gz)

joint

cartesian

plane

sphere

axis_1

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Documentation update

## Checklist

- [x] I have run the `black .` command to format the code base.
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Dependencies have been updated, if applicable.

### CODE REVIEW

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py [hunk 1]
```diff
@@ -0,0 +1,1480 @@
...
+                # Create bounds tensor: [[x_min, x_max], [y_min, y_max], [z_min, z_max]]
+                cartesian_bounds = torch.stack([min_bounds, max_bounds], dim=1)
+
+                # Format bounds with better precision and units
+                min_bounds_np = min_bounds.cpu().numpy()
+                max_bounds_np = max_bounds.cpu().numpy()
+
+                # Calculate workspace dimensions for additional context
+                dimensions = max_bounds_np - min_bounds_np
+                volume = np.prod(dimensions)
```
[the remark this answers is not in the document]
Comment (chase6305) [reply embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/DexForce/EmbodiChain/pull/14#discussion_r2571295675)

**Comment by chase6305** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/DexForce/EmbodiChain/pull/14#pullrequestreview-3518516676)

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py [hunk 2]
```diff
@@ -0,0 +1,1478 @@
...
+
+    Analyzes the reachable workspace of a robot by sampling joint configurations,
+    computing forward kinematics, and generating metrics and visualizations.
+    """
+
+    def __init__(
+        self,
+        robot: Robot,
+        config: Optional[WorkspaceAnalyzerConfig] = None,
+        control_part_name: Optional[str] = None,
```
Comment (yuecideng) [inline embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: Why not put `control_part_name` into Config?

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py [hunk 3]
```diff
@@ -0,0 +1,1478 @@
...
+        sim_manager: Optional[SimulationManager] = None,
+        device: torch.device = torch.device("cpu"),
```
Comment (yuecideng) [inline embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: device should be consistent with device in sim manager. So it can be removed here (just use sim.device inside)

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py [hunk 4]
```diff
@@ -0,0 +1,1478 @@
...
+            # User explicitly specified a control part
+            logger.log_info(f"Using user-specified control part: {control_part_name}")
+            return control_part_name
+
+        # Try to find a suitable default control part
+        if hasattr(self.robot, "cfg") and hasattr(self.robot.cfg, "control_parts"):
+            control_parts = self.robot.cfg.control_parts
+            if control_parts:
+                # Priority order for default control parts
+                priority_parts = ["left_arm", "right_arm"]
```
Comment (yuecideng) [inline embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: Default variable can be defined as class variable to hightlight it is a gobal (under context of this class) default.

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py [hunk 5]
```diff
@@ -0,0 +1,1478 @@
...
+            robot: Robot instance to analyze.
+            config: Configuration object. If None, uses defaults.
+            control_part_name: Name of the control part (e.g., "left_arm", "right_arm").
+                              If None, uses the default solver or first available control part.
+            sim_manager: SimulationManager instance. Defaults None.
+            device: PyTorch device for computations.
+        """
+        self.robot = robot
+        self.config = config or WorkspaceAnalyzerConfig()
+        self.sim_manager = sim_manager
```
Comment (yuecideng) [inline embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py:175]: Currently, `WorkspaceAnalyzer` is designed for `num_envs=1`, so we should add protect and check whether the current setting is acceptable.

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py [hunk 6]
```diff
@@ -0,0 +1,1478 @@
...
+                return first_part
+
+        # Fall back to None (will use default solver)
+        logger.log_info("No specific control part specified, using default solver")
+        return None
+
+    def _setup_joint_limits(self) -> None:
+        """Extract and setup joint limits from the robot."""
+        # Get all joint limits from robot
+        all_joint_limits = self.robot._entities[0].get_joint_limits()
```
Comment (yuecideng) [inline embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: Use `self.robot.body_data.qpos_limits.squeese(0)` or `self.robot.body_data.qpos_limits[0]`

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py (hunk 2 above)
Comment (chase6305) [reply embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: Done.

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py (hunk 3 above)
Comment (chase6305) [reply embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: Use sim_manager's device if available, otherwise default to CPU

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py (hunk 4 above)
Comment (chase6305) [reply embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: Done. Using `DEFAULT_CONTROL_PART_PRIORITY = ["left_arm", "right_arm"]`.

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py (hunk 5 above)
Comment (chase6305) [reply embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py:175]: I added `_check_num_envs_compatibility` to indicate that the current module only targets one or the first environment. (Comments related to `sim_manager` and `robot` are also included.)

File: embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py (hunk 6 above)
Comment (chase6305) [reply embodichain/lab/sim/utility/workspace_analyzer/workspace_analyzer.py]: Done. Using `self.robot.body_data.qpos_limits[0]`.

**Review by yuecideng — APPROVED** [review]
